### DISCUSSION TITLE
YAML template in Go template

### DISCUSSION THREAD

**Discussion by mhimmiche** [helmfile/helmfile#1034] [Questions]
Hi all, hoping someone might have come across the issue I'm seeing. I've looked through the docs and code and can't quite spot what I'm doing wrong. I'm also using this as a ref - https://lyz-code.github.io/blue-book/devops/helmfile/#yaml-templates-in-go-templates

Anyways, I'm trying to create a bunch of configmaps out of json files (they're Grafana dashboards). Unfortunately, sometimes they contain `{{ stuff }}` or `{{stuff}}`.  I'm using the official Grafana helm chart, this is what I have so far
```
{{ $env_name := .Environment.Name }}
{{ $path := list "../../" $env_name "/config/grafana/dashboards" | join "" }}
extraObjects:
{{ range $_, $item := readDir $path }}
{{- $itemSplit := splitList  "/" $item -}}
{{- $itemValue := $itemSplit | last -}}
{{- $fileName := splitList "." $itemValue | first -}}
- apiVersion: v1
  kind: ConfigMap
  metadata:
    name: dashboard-{{ $fileName }}
    labels:
      dashboard-provider: grafana-sc-dashboard
  data:
    {{ $fileName }}.json: |-
{{ readFile $item | indent 10 }}
{{ end }}
```
I've tried everything, from piping toJson, toString, changing the indentation, what have you, but I still get the following error
```
parse error at (grafana/templates/extra-manifests.yaml:2483): function "instance" not defined
```
Because the json has `{{instance}}`. I can't quite figure out how to get it working and I'm a bit at a loss here.

**Comment by yxxhero** [comment]
{{ instance }}? I can't find the code.

**Reply by mhimmiche** [reply]
The string {{instance}} is in the json I’m parsing

**Comment by mhimmiche** [comment]
nvm, I chased down the downstream repo and noticed that it ran `tpl (toYaml .) $` 🤦
